src/main/java/uk/gov/hmcts/reform/iacaseapi/domain/handlers/presubmit/FeePaymentHandler.java

```diff
@@ -61,7 +60,83 @@ public boolean canHandle(
             throw new IllegalStateException("Cannot handle callback");
         }
 
-        AsylumCase asylumCase = feePayment.aboutToSubmit(callback);
+        AsylumCase asylumCase =
+            callback
+                .getCaseDetails()
+                .getCaseData();
+
+        AppealType appealType = asylumCase.read(APPEAL_TYPE, AppealType.class)
+            .orElseThrow(() -> new IllegalStateException("Appeal type is not present"));
+
+        YesOrNo isRemissionEnabled = asylumCase.read(IS_REMISSIONS_ENABLED, YesOrNo.class)
+            .orElse(YesOrNo.NO);
+
+        switch (appealType) {
+            case EA:
+            case HU:
+            case PA:
+                Optional<RemissionType> optRemissionType = asylumCase.read(REMISSION_TYPE, RemissionType.class);
+                if (isRemissionEnabled == YesOrNo.YES
+                    && optRemissionType.isPresent()
+                    && optRemissionType.get() == RemissionType.HO_WAIVER_REMISSION) {
+
+                    setFeeRemissionTypeDetails(asylumCase);
+                } else {
+                    asylumCase = feePayment.aboutToSubmit(callback);
+                    setFeePaymentDetails(asylumCase, appealType);
+                }
+                asylumCase.clear(RP_DC_APPEAL_HEARING_OPTION);
+                break;
+
+            case DC:
+            case RP:
+                String hearingOption = asylumCase.read(RP_DC_APPEAL_HEARING_OPTION, String.class)
+                    .orElseThrow(() -> new IllegalStateException("Appeal hearing option is not present"));
+                asylumCase.write(DECISION_HEARING_FEE_OPTION, hearingOption);
```

Comment:
Test if this written for all appeal_types

Reply:
Added the assertion for DC/RP appeal types